app/services/product_filter_service.rb

```diff
@@ -0,0 +1,41 @@
+class ProductFilterService
+  def initialize(params)
+    @params = params
+  end
+
+  def filter
+    products = Product.all
+    
+    if @params[:category].present? && @params[:category] != 'Category'
```

Comment:
Similar to the comment below in vegetarian. Is there any chance you'll be sending the parameter as `Category`? You might want the check to be more specific, so you would want to check if `@params[:category]` is one of the permitted values (in this case, the values of the enum).

So a possible solution for this would be:
```ruby
if @params[:category].present? && @params[:category].in?(Product.categories.keys)
```

You can also check this is Rails console, but `Product.categories` should return the enum values and `Product.categories.keys` will return the strings that match the enum values